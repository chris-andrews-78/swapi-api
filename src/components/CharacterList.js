import React, { useContext } from 'react'
import {SwapiContext} from "../context/SwapiAPI";

const CharacterList = () =>{

    const swapiSearch = useContext(SwapiContext)

    //User can select character from a list if multiple are returned so call context API function to do this
    function handleClick(character) {
        swapiSearch.setCharacter(character);
    }

    return(
        <div>
          {(swapiSearch.characters.length > 1) ? (
            <div className="sw-app-card">
                <p>We found multiple characters with your search, click on the name to display the profile.</p>
                <ul>
                {swapiSearch.characters.map((character) => (
                    <li key={character.url}><a href="#" onClick={() => {handleClick(character)}}>{character.name}</a></li>
                ))}
                </ul>
            </div>
          ) : ""}
        </div>
    )
}

export default CharacterList;