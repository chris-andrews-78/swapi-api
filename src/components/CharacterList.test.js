import React from 'react';
import renderer from 'react-test-renderer';
import CharacterList from './CharacterList';
import SwapiProvider from "../context/SwapiAPI";

test('Test Character List', () => {

    const component = renderer.create(
        <SwapiProvider>  
            <CharacterList />
        </SwapiProvider>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

});