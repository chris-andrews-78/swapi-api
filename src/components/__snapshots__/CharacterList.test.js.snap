// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Character List 1`] = `
<div>
  
</div>
`;
